src/web/mjs/connectors/PixivComics.mjs

```diff
@@ -76,13 +76,141 @@ export default class PixivComics extends Connector {
 
     async _getPages(chapter) {
         const timestamp = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
-        const hash = CryptoJS.MD5(timestamp + 'mAtW1X8SzGS880fsjEXlM73QpS1i4kUMBhyhdaYySk8nWz533nrEunaSplg63fzT').toString(CryptoJS.enc.Hex);
-        const uri = new URL(`episodes/${chapter.id}/read`, this.apiURL);
+        const hash = CryptoJS.SHA256(timestamp + 'mAtW1X8SzGS880fsjEXlM73QpS1i4kUMBhyhdaYySk8nWz533nrEunaSplg63fzT').toString(CryptoJS.enc.Hex);
+        const uri = new URL(`episodes/${chapter.id}/read_v3`, this.apiURL);
         const request = new Request(uri, this.requestOptions);
         request.headers.set('x-requested-with', 'pixivcomic');
         request.headers.set('x-client-time', timestamp);
         request.headers.set('x-client-hash', hash);
         const data = await this.fetchJSON(request);
-        return data.data.reading_episode.pages.map(image => this.createConnectorURI(image.url));
+        return data.data.reading_episode.pages.map(image => this.createConnectorURI(image));
     }
-}
\ No newline at end of file
+
+    async _handleConnectorURI(payload) {
+        let request = new Request(payload.url, this.requestOptions);
+        request.headers.set('X-Cobalt-Thumber-Parameter-GridShuffle-Key', payload.key);
+        const res = await fetch(request);
+        const blob = await res.blob();
+        const canvas = await this.descrambleImage(blob, payload);
+        const blobFinally = await this._canvasToBlob(canvas);
+        return this._blobToBuffer(blobFinally);
+    }
+
+    async descrambleImage(blob, payload) {
+        const image = await createImageBitmap(blob);
+        const canvas = document.createElement('canvas');
+        canvas.width = payload.width;
+        canvas.height = payload.height;
+        let ctx = canvas.getContext( '2d' );
+
+        ctx.drawImage(image, 0, 0);
+        const i = ctx.getImageData(0, 0, payload.width, payload.height);
+        const l = await this.descrambleData(i.data, 4, payload.width, payload.height, payload.gridsize, payload.gridsize, '4wXCKprMMoxnyJ3PocJFs4CYbfnbazNe', payload.key, true );
+        const o = new ImageData(l, payload.width, payload.height);
+        ctx.putImageData(o, 0, 0);
+        return canvas;
+    }
+
+    async descrambleData(e, t, i, r, n, s, a, l, o) {
```

Comment:
This code seems copy & pasted from the website, may move this code outside of this class (near `PixivShuffler` class declaration)